### ISSUE TITLE
Called twice with react strict mode

### ISSUE THREAD

**Issue by shadowvzs** [Andarist/use-constant#10]
Hi, in strict mode it is called twice, which is an issue if we have an api call method or class instantiation

**Comment by Andarist** [comment]
This is how React works - you can't instantiate anything that has side-effects with `useConstant`. Side effects belong to `useEffect`/event handlers (or to outside of your components)

**Comment by shadowvzs** [comment]
@Andarist I thought it should be used for instantiating a store class too with useConstant but this isn't the case, useEffect is called after the component was mounted and not parallel

**Comment by Andarist** [comment]
Yes, but again - this is how React works. `useConstant` is resolved in render, and render should always be pure.
